kernel/src/listed_log_files.rs

```diff
@@ -621,4 +636,51 @@ mod list_log_files_with_log_tail_tests {
         assert_source(&result[1], CommitSource::Catalog);
         assert_source(&result[2], CommitSource::Catalog);
     }
+
+    #[test]
+    fn test_listing_omits_staged_commits() {
```

Comment:
Clarifying question: Is this testing the case of: while listing the `_delta_log` directory, you come across staged raitifed commit files? are these `_delta_log/0000N.uuid.json`? or `_delta_log/_staged_commits/0000N.uuid.json`?

Reply:
TLDR: this is just test case that we don't LIST `_staged_commits`.

they are `_delta_log/_staged_commits/0000N.uuid.json` to answer that first question 👍 

> And is the "LIST" engine API expected to list only files in that single directory? or recursively list subfolders too?

AFAIK current semantics are listing subdirs (I _think_ this is just S3/etc. default?)

> I ask because: if you ever see _delta_log/000N.uuid.json --> that is a corrupted table. Here we are ommitting them -- but I think we should be throwing an exception if I understand this test setup correctly?

Hm we do have a test (not this one) but we silently ignore them. It would just be interpreted as an Unknown path in the log dir.